cmd/testpeers.go

```diff
@@ -63,7 +100,150 @@ func supportedSelfTestCases() map[testCaseName]func(context.Context, *testPeersC
+		}
+
+		p, err := p2p.NewPeerFromENR(record, i)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		peers = append(peers, p)
+	}
+
+	key, err := p2p.LoadPrivKey(cfg.DataDir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	r, err := enr.New(key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mePeer, err := p2p.NewPeerFromENR(r, len(cfg.ENRs))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Info(ctx, "Self p2p name resolved", z.Any("name", mePeer.Name))
+
+	peers = append(peers, mePeer)
+
+	allENRs := cfg.ENRs
+	allENRs = append(allENRs, r.String())
+	slices.Sort(allENRs)
+	allENRsString := strings.Join(allENRs, ",")
+	h := sha256.New()
+	_, err = h.Write([]byte(allENRsString))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "calculate sha256")
+	}
+	allENRsHash := h.Sum(nil)
+
+	return setupP2P(ctx, key, cfg.P2P, peers, allENRsHash)
+}
+
+func setupP2P(ctx context.Context, key *k1.PrivateKey, cfg p2p.Config, peers []p2p.Peer, enrsHash []byte) (host.Host, func(), error) {
+	var peerIDs []peer.ID
+	for _, p := range peers {
+		peerIDs = append(peerIDs, p.ID)
+	}
+
+	if err := p2p.VerifyP2PKey(peers, key); err != nil {
+		return nil, nil, err
+	}
+
+	relays, err := p2p.NewRelays(ctx, cfg.Relays, hex.EncodeToString(enrsHash))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	connGater, err := p2p.NewConnGater(peerIDs, relays)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	tcpNode, err := p2p.NewTCPNode(ctx, cfg, key, connGater, false)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	p2p.RegisterConnectionLogger(ctx, tcpNode, peerIDs)
+
+	for _, relay := range relays {
+		relay := relay
+		go p2p.NewRelayReserver(tcpNode, relay)(ctx)
+	}
+
+	go p2p.NewRelayRouter(tcpNode, peerIDs, relays)(ctx)
+
+	// Register peerinfo server handler for identification to relays (but do not run peerinfo client).
+	gitHash, _ := version.GitCommit()
+	_ = peerinfo.New(tcpNode, peerIDs, version.Version, enrsHash, gitHash, nil, false)
+
+	return tcpNode, func() {
+		_ = tcpNode.Close()
+	}, nil
+}
+
+func pingPeerOnce(ctx context.Context, tcpNode host.Host, peer p2p.Peer) (ping.Result, error) {
+	select {
+	case <-ctx.Done():
+		return ping.Result{}, errors.New("context done")
+	default:
+		pingSvc := ping.NewPingService(tcpNode)
+		pingCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		for {
+			pingChan := pingSvc.Ping(pingCtx, peer.ID)
+			select {
+			case <-pingCtx.Done():
+				return ping.Result{}, errors.New("context done")
+			case result := <-pingChan:
+				return result, nil
+			}
+		}
+	}
+}
+
+func pingPeerContinuously(ctx context.Context, tcpNode host.Host, peer p2p.Peer, resCh chan ping.Result) {
+	for {
+		select {
+		case <-ctx.Done():
+			select {
+			case <-resCh:
+				return
+			default:
+				close(resCh)
+				return
+			}
+		default:
+			r, err := pingPeerOnce(ctx, tcpNode, peer)
+			if err != nil {
+				return
+			}
+			resCh <- r
+			nBig, err := rand.Int(rand.Reader, big.NewInt(100))
```

Comment:
Big integers seems overkill here, why not just https://pkg.go.dev/math/rand#Intn?

Reply:
Agree, that's how I started, but linter complained... wanted me to use `crypto/rand`.

`G404: Use of weak random number generator (math/rand instead of crypto/rand) (gosec)`

I may override it for this instance here

Reply:
Yes I think for this instance it is better to override.